flaml/automl.py

```diff
@@ -2382,24 +2362,52 @@ def _search_parallel(self):
         else:
             self._state.time_from_start = time.time() - self._start_time_flag
             time_left = self._state.time_budget - self._state.time_from_start
-            search_alg = SearchAlgo(
-                metric="val_loss",
-                space=space,
-                low_cost_partial_config=self.low_cost_partial_config,
-                points_to_evaluate=self.points_to_evaluate,
-                cat_hp_cost=self.cat_hp_cost,
-                resource_attr=self.resource_attr,
-                min_resource=self.min_resource,
-                max_resource=self.max_resource,
-                config_constraints=[
-                    (partial(size, self._state), "<=", self._mem_thres)
-                ],
-                metric_constraints=self.metric_constraints,
-                seed=self._seed,
-                time_budget_s=time_left,
-            )
+            if self._hpo_method != "optuna":
+                search_alg = SearchAlgo(
+                    metric="val_loss",
+                    space=space,
+                    low_cost_partial_config=self.low_cost_partial_config,
+                    points_to_evaluate=self.points_to_evaluate,
+                    cat_hp_cost=self.cat_hp_cost,
+                    resource_attr=self.resource_attr,
+                    min_resource=self.min_resource,
+                    max_resource=self.max_resource,
+                    config_constraints=[
+                        (partial(size, self._state), "<=", self._mem_thres)
+                    ],
+                    metric_constraints=self.metric_constraints,
+                    seed=self._seed,
+                    time_budget_s=time_left,
+                )
+            else:
+                converted_space = SearchAlgo.convert_search_space(space)
+
+                removed_keys = set(space.keys()).difference(converted_space.keys())
+                new_points_to_evaluate = []
+                for idx in range(len(self.points_to_evaluate)):
+                    r = self.points_to_evaluate[idx].copy()
+                    for each_key in removed_keys:
+                        r.pop(each_key)
+                    new_points_to_evaluate.append(r)
+
+                search_alg = SearchAlgo(
+                    metric="val_loss",
+                    mode="min",
+                    points_to_evaluate=[
+                        p
+                        for p in new_points_to_evaluate
+                        if len(p) == len(converted_space)
+                    ],
+                )
             search_alg = ConcurrencyLimiter(search_alg, self._n_concurrent_trials)
         resources_per_trial = self._state.resources_per_trial
+
+        ray_args = (
+            self._state.fit_kwargs.get("use_ray")
+            if isinstance(self._state.fit_kwargs.get("use_ray"), dict)
```

Comment:
"use_ray" is not in fit_kwargs any more, right?

Reply:
This is for passing the customized keyword args to ray, e.g., local_dir

Reply:
"use_ray" is not in `fit_kwargs`.